compiler/src/dmd/expression.d

```diff
@@ -4852,6 +4852,12 @@ extern (C++) final class EqualExp : BinExp
     {
         v.visit(this);
     }
+
+    /// Returns: `true` when this expression compares equality (==), `false` for inequality (!=)
+    extern (D) bool isPositive() @safe pure const
```

Comment:
I'm not sure this is what I would call this function.
For  `op == EXP.notEqual`, I would call `IsNegated`

3 of its 5 uses are of the form `e.isPositive ? EXP.identity : EXP.notIdentity`, I'd rather rewrite that as a function

Reply:
I called it `isNegative` first, but then you'd get double negations (`!e.isNegative`). I'll put off that change for now.